designcompose/src/main/java/com/android/designcompose/DesignText.kt

```diff
@@ -318,7 +310,7 @@ internal fun DesignText(
                 )
             } else {
                 // Text needs to use a modifier that sets the size so that it wraps properly
-                val height = renderHeight ?: layout?.height() ?: 0
+                val height = layout?.height() ?: 0
                 val textModifier = modifier.sizeToModifier(layout?.width() ?: 0, height)
                 BasicText(
```

Comment:
Can this take `ParagraphIntrinsics` instead of `annotatedText`? I think it's going to make a `ParagraphIntrinsics` internally somewhere, and that seems slow.

Reply:
`BasicText()` doesn't take `ParagraphIntrinsics` -- am I missing something?

Reply:
No, you're not, I was assuming that there would be a `BasicText` constructor that took `ParagraphIntrinsics` instead of an `AnnotatedString` and a `TextStyle`... but there doesn't appear to be. Looking at the implementation, `BasicText` eventually uses this [`MultiParagraphLayoutCache`](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/modifiers/MultiParagraphLayoutCache.kt) which keeps a `MultiParagraphIntrinsics` instance around for the text and style that's passed in.

So Compose uses `MultiParagraph` and `MultiParagraphIntrinsics` instead of `Paragraph` and `ParagraphIntrinsics`, because a `BasicText` supports multiple paragraph styles in a run. We don't encounter that, so `Paragraph` and `ParagraphIntrinsics` is OK for us, but unfortunately I don't see any way for us to use Compose's `MultiParagraphLayoutCache` -- so while we use `BasicText`, we're always going to be laying out text once for ourselves, and once for Compose. (So what you have now is as good as it gets).